src/html2ans/parsers/text.py

```diff
@@ -266,66 +296,92 @@ class ListParser(BaseElementParser):
                 },
                 {
-                    "type": "text",
-                    "content": "Unparalleled reporting."
+                    'type': 'list',
+                    'list_type': 'ordered',
+                    'items': [
+                        {
+                            'type': 'text',
+                            'content': 'Unparalleled reporting.'
+                        },
+                        {
+                            'type': 'text',
+                            'content': 'Expert insight.'
+                        }
+                    ]
                 },
                 {
-                    "type": "text",
-                    "content": "Expert insight."
-                },
-                {
-                    "type": "text",
-                    "content": "Clear analysis."
+                    'type': 'text',
+                    'content': '<p>Clear analysis.</p>'
                 }
             ]
         }
 
+
     """
     applicable_elements = ['ul', 'ol']
+    text_parser = ListItemParser
 
     def parse(self, element, *args, **kwargs):
+        list_elements = []
+
+        def _add_list_element(parsed_list_item, check_field):
+
+            if isinstance(parsed_list_item, dict) and parsed_list_item.get(check_field):
+                list_elements.append(parsed_list_item)
+
         if element.name == 'ul':
             list_type = 'unordered'
         else:
             list_type = 'ordered'
-        list_elements = []
+
         for list_item in element.contents:
+
             # as of ANS 0.6.2, has to be either text or another list
             # https://github.com/washingtonpost/ans-schema/blob/master/src/main/resources/schema/ans/0.6.2/story_elements/list_element.json
             if isinstance(list_item, Tag) and list_item.contents:
-                # we could do something complicated so that this parser has its own sub-parsers
-                # but for the time being, assume the list items are fairly simple
-                # if this item is wrapping something else, just take the first child
-                list_item = list_item.contents[0]
-            if isinstance(list_item, Tag) and list_item.name in self.applicable_elements:
-                parsed_list_item = self.parse(list_item).output
-                field_check = "items"
-            else:
-                parsed_list_item = ParagraphParser().parse(list_item).output
-                field_check = "content"
-            if isinstance(parsed_list_item, dict) and parsed_list_item.get(field_check):
-                list_elements.append(parsed_list_item)
+                child_lists = list_item.find_all(self.applicable_elements)
+
+                if child_lists:
+
+                    for child in child_lists:
+                        _add_list_element(self.parse(child).output, "items")
+
+                else:
+
+                    if self.text_parser().is_applicable(list_item):
```

Comment:
Let's make the `text_parser` an attribute that's set by the constructor named `list_item_parser`. If no `list_item_parser` is passed, set it to `ListItemParser()`. That way, we won't create two separate objects here.